README.md

```diff
@@ -654,6 +654,50 @@ app.use('/notify', handler);
 
 For example of the notifications payload [see details](./notifications.md)
 
+### smartNotificationHandler() `BETA`
+
+`smartNotificationHandler` is a next generation `notifHandler` cooperating with our internal notification tool. To mark connector as using the smartNotifier `smart-notifier` tag should be present in `manifest.json` file:
+
+```json
+{
+  "tags": ["smart-notifier"],
+  "subscriptions": [{
+    "url" : "/notify"
+  }]
+}
+```
+
+Here's how to use it.
+
+```js
+import { smartNotificationHandler } from "hull/lib/utils";
+const app = express();
+
+const handler = smartNotificationHandler({
+  handlers: {
+    "ship:update": function(ctx, messages = []) {},
+    "segment:update": function(ctx, messages = []) {},
+    "segment:delete": function(ctx, messages = []) {},
+    "user:update": function(ctx, messages = []) {
+      console.log("Event Handler here", ctx, messages);
```

Comment:
Need an example of response with flow control

Reply:
Added it here and more documentation below.